libraries/Ticker/Ticker.h

```diff
@@ -48,8 +42,8 @@ class Ticker
 
 	void attach(float seconds, callback_function_t callback)
 	{
-		_callback_function = callback;
-		attach(seconds, _static_callback, (void*)this);
+		_callback_function = std::move(callback);
+		_attach_ms(seconds * 1000, true, _static_callback, this);
```

Comment:
One of the reasons the previous code was routed through attach is that the ESP doesn't have hw floats. You unraveled that to route all calls to the new private attach ms, but in so doing you have to multiply a float in several cases, which translates to the compiler implementing that multiplication in sw several times over. I'm not sure which is better. It might be a good idea to wrap the multiplication in a function. I'll leave the decision to you.

Reply:
Thinking about this, the whole concept of using float of all datatypes to encode seconds - I surmise with the intention of supplying split seconds - is rather broken for an MCU. Is there any other API that does this? I only ever see µs and ms. os_timer_arm expects ms, there is an option to switch to µs, each with 5ms or 100µs minimum, so float is just misleading - if there's a performance impact, I am inclined rather to remove float altogether rather than spend my time on that.

Reply:
Anyway, done - by adding a _attach_s() function as single spot where float to uint32_t conversion takes place.